version: '3'

services:
  traefik:
    image: traefik:v2.4
    networks:
      - proxy
    ports:
      - 80:80
      - 443:443
    command:
      - --providers.docker
      - --providers.docker.swarmmode
    environment:
      - CF_API_EMAIL=<TODO>
      - CF_API_KEY=<TODO>
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./data/traefik.yml:/traefik.yml:ro
      - ./data/acme.json:/acme.json
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.entrypoints=https"
      - "traefik.http.routers.traefik.tls=true"
      - "traefik.http.routers.traefik.tls.certresolver=cloudflare"
      - "traefik.http.routers.traefik.tls.domains[0].main=<TODO>"
      - "traefik.http.routers.traefik.tls.domains[0].sans=*.<TODO>"
      - "traefik.http.middlewares.traefik-auth.basicauth.users=<TODO>"
      - "traefik.http.middlewares.sslheader.headers.customrequestheaders.X-Forwarded-Proto=https"
      - "traefik.http.routers.traefik.rule=Host(`<TODO>`)"
      - "traefik.http.routers.traefik.middlewares=traefik-auth"
      - "traefik.http.routers.traefik.service=api@internal"
    deploy:
      mode: global
      restart_policy:
        condition: on-failure
      placement:
        constraints:
          - node.role == manager


networks:
  proxy:
    external: true